sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -289,6 +290,11 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       checkStringToTimestamp("", None)
       checkStringToTimestamp("    ", None)
       checkStringToTimestamp("+", None)
+      checkStringToTimestamp("T", None)
+      checkStringToTimestamp("2015-03-18T", None)
```

Comment:
Maybe we can do some special handling to allow this.

Reply:
Is this valid? doesn't quite look like it

Reply:
Personally I think it's invalid. But since previously it's allowed, so maybe someone will argue we should still allow it

Reply:
Was it previously allowed? this test is new

Reply:
It was previously allowed: https://github.com/apache/spark/pull/33490/files#diff-95555ac355a4972b9a61cad3e6599b0ac3c242877c1a393ecc5a77842e90091eL185